foursight_core/app_utils.py

```diff
@@ -1901,36 +1935,42 @@ def is_non_empty_json_file(file: str) -> bool:
             try:
                 if os.path.exists(file):
                     if os.stat(file).st_size > 0:
-                        with io.open(file, "r") as f:
-                            content = json.load(f)
-                            return (isinstance(content, list) or isinstance(content, dict)) and len(content) > 0
+                        return True
             except Exception:
                 pass
             return False
 
-        if not base_dir:
-            base_dir = os.path.dirname(__file__)
-        check_setup_file = None
-        if os.environ.get("CHALICE_LOCAL") == "1":
-            check_setup_dir = os.path.normpath(os.path.join(base_dir, "../chalicelib_local"))
-            check_setup_file = os.path.join(check_setup_dir, AppUtilsCore.CHECK_SETUP_FILE_NAME)
-            if not is_non_empty_json_file(check_setup_file):
-                check_setup_file = None
-        if not check_setup_file:
-            check_setup_dir = os.environ.get("FOURSIGHT_CHECK_SETUP_DIR", "")
-            check_setup_file = os.path.join(check_setup_dir, AppUtilsCore.CHECK_SETUP_FILE_NAME)
-            if not is_non_empty_json_file(check_setup_file):
-                check_setup_file = None
-        if not check_setup_file:
-            check_setup_dir = base_dir
-            check_setup_file = os.path.join(check_setup_dir, AppUtilsCore.CHECK_SETUP_FILE_NAME)
-            if not is_non_empty_json_file(check_setup_file):
-                check_setup_file = None
-        return check_setup_file
+        config_file = None
+        chalicelib_dir = None
+        if ROUTE_CHALICE_LOCAL:
+            if not chalicelib_dir:
+                chalicelib_dir = _get_chalicelib_dir()
+            config_dir = os.path.normpath(os.path.join(chalicelib_dir, "../chalicelib_local"))
+            config_file = os.path.join(config_dir, file_name)
+            if not is_non_empty_file(config_file):
+                config_file = None
+        if not config_file:
+            env_based_dir = os.environ.get("FOURSIGHT_CHECK_SETUP_DIR", "")
+            config_file = os.path.join(env_based_dir, file_name)
+            if not is_non_empty_file(config_file):
+                config_file = None
+        if not config_file:
+            if not chalicelib_dir:
+                chalicelib_dir = _get_chalicelib_dir()
+            config_file = os.path.join(chalicelib_dir, file_name)
+            if not is_non_empty_file(config_file):
+                config_file = None
+        if not config_file:
+            foursight_core_dir = os.path.dirname(__file__)
+            config_file = os.path.join(foursight_core_dir, file_name)
+            if not is_non_empty_file(config_file):
+                config_file = None
+        return config_file
 
     _singleton = None
+
     @staticmethod
-    def singleton(cls = None):
+    def singleton(cls=None):
```

Comment:
Nothing wrong with this, but I still recommend not calling the arg `cls` because it makes it look like a class method. Maybe call it `specific_class` or something that looks like it's meant to be passed explicitly.

I'm not even sure I'm happy about it being defaulted to `AppUtilsCore` mostly because I think the caller should know what class they're intending.

Reply:
Yeah that's better.